src/eventsub/channel/chat/notification.rs

```diff
@@ -0,0 +1,541 @@
+    const NAME: &'static str = "sub";
+}
+impl crate::eventsub::NamedField for Resubscription {
+    const NAME: &'static str = "resub";
+}
+impl crate::eventsub::NamedField for SubGift {
+    const NAME: &'static str = "sub_gift";
+}
+impl crate::eventsub::NamedField for CommunitySubGift {
+    const NAME: &'static str = "community_sub_gift";
+}
+impl crate::eventsub::NamedField for GiftPaidUpgrade {
+    const NAME: &'static str = "gift_paid_upgrade";
+}
+impl crate::eventsub::NamedField for PrimePaidUpgrade {
+    const NAME: &'static str = "prime_paid_upgrade";
+}
+impl crate::eventsub::NamedField for Raid {
+    const NAME: &'static str = "raid";
+}
+impl crate::eventsub::NamedField for Unraid {
+    const NAME: &'static str = "unraid";
+}
+impl crate::eventsub::NamedField for PayItForward {
+    const NAME: &'static str = "pay_it_forward";
+}
+impl crate::eventsub::NamedField for Announcement {
+    const NAME: &'static str = "announcement";
+}
+impl crate::eventsub::NamedField for CharityDonation {
+    const NAME: &'static str = "charity_donation";
+}
+impl crate::eventsub::NamedField for BitsBadgeTier {
+    const NAME: &'static str = "bits_badge_tier";
+}
+
+/// A badge
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "deny_unknown_fields", serde(deny_unknown_fields))]
+#[non_exhaustive]
+pub struct Badge {
+    /// An ID that identifies this set of chat badges. For example, Bits or Subscriber.
+    pub set_id: types::BadgeSetId,
+    /// An ID that identifies this version of the badge. The ID can be any value. For example, for Bits, the ID is the Bits tier level, but for World of Warcraft, it could be Alliance or Horde.
+    pub id: types::ChatBadgeId,
+    /// Contains metadata related to the chat badges in the badges tag. Currently, this tag contains metadata only for subscriber badges, to indicate the number of months the user has been a subscriber.
+    pub info: String,
+}
+
+/// A message
+// XXX: this struct can never be deny_unknown_fields
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[non_exhaustive]
+pub struct Message {
+    /// The chat message in plain text.
+    pub text: String,
+    /// Ordered list of chat message fragments.
+    pub fragments: Vec<Fragment>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[non_exhaustive]
+pub struct Fragment {
+    /// Message text in fragment
+    pub text: String,
+    #[serde(flatten)]
+    pub data: FragmentData,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[non_exhaustive]
+#[serde(tag = "type", rename_all = "lowercase")]
+pub enum FragmentData {
+    /// A Cheermote.
+    #[serde(with = "crate::eventsub::enum_field_as_inner")]
+    Cheermote(Cheermote),
+    /// A Emote.
+    #[serde(with = "crate::eventsub::enum_field_as_inner")]
+    Emote(Emote),
+    /// A Mention.
+    #[serde(with = "crate::eventsub::enum_field_as_inner")]
+    Mention(Mention),
+}
```

Comment:
this will probably (almost 100% confirmed, totally not a leek) be used for `channel.chat.message`